subprojects/performance/src/templates/archivePerformanceProject/build.gradle

```diff
@@ -0,0 +1,43 @@
+/*
+* Copyright 2017 the original author or authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+task zip(type: Zip) {
+    from 'build.gradle'
+}
+
+task visitZip(dependsOn: zip) {
+    doLast {
+        (1..10000).each {
+            zipTree(zip.archivePath).each {
+                logger.info it.path
+            }
+        }
+    }
+}
+
+task tar(type: Tar) {
+    from 'build.gradle'
+}
+
+task visitTar(dependsOn: tar) {
+    doLast {
+        (1..10000).each {
+            tarTree(tar.archivePath).each {
+                logger.info it.path
+            }
+        }
+    }
+}
```

Comment:
It would make the test more realistic if we had more files in the archive, and fewer iterations. It would ensure for example that the TAR is not read from the beginning every time we extract a single entry.